frontend/src/components/common/menu.jsx

```diff
@@ -8,54 +9,77 @@ export default class Menu extends React.Component {
         super(props);
         this.toggle = this.toggle.bind(this);
         this.close = this.close.bind(this);
-        this.handleToggleClick = this.handleToggleClick.bind(this);
-        this.handleMenuClick = this.handleMenuClick.bind(this);
+        this.onClickOutside = this.onClickOutside.bind(this);
         this.state = { show: false };
     }
 
     componentDidMount() {
-        document.addEventListener('click', this.close, false);
+        document.addEventListener('click', this.onClickOutside, false);
     }
 
     componentWillUnmount() {
-        document.removeEventListener('click', this.close, false);
+        document.removeEventListener('click', this.onClickOutside, false);
     }
 
     toggle() {
-        this.setState({ show: !this.state.show });
+        const show = !this.state.show;
+
+        this.setState({ show });
+
+        if(show && this.props.onOpen) {
+            this.props.onOpen();
+        }
     }
 
     close() {
         this.setState({ show: false });
     }
 
-    handleToggleClick(event) {
-        this.toggle();
-        event.nativeEvent.stopImmediatePropagation();
-    }
+    onClickOutside(ev) {
+        if(this.state.show && this.ref) {
+            const contains = this.ref.contains(ev.target);
 
-    handleMenuClick(event) {
-        this.close();
-        event.nativeEvent.stopImmediatePropagation();
+            if(!contains) {
+                this.close();
+            }
+        }
     }
 
     getMenu() {
         const items = this.props.options.map((option, index) => {
-            return <li key={index} onClick={ option.action }>{ option.label }</li>
+            if(React.isValidElement(option)) {
```

Comment:
Out of interest, what are you testing for here?

Reply:
Ah I re-used this menu class by injecting the checkbox filters as React elements for each option from the parent component rather than text options